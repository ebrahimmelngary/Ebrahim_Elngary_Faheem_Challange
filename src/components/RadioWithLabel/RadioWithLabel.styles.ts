
import { StyleSheet } from 'react-native';
import { COLORS } from '../../common/colors';

export default StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
  },
  radioButton: {
    width: 25,
    height: 25,
    marginHorizontal: 10,
    borderRadius: 25,
    borderWidth: 3,
    alignItems: 'center',
    justifyContent: 'center',
    borderColor: COLORS.white,
  },
  activeRadioButton: {
    width: 15,
    height: 15,
    borderRadius: 15,
    backgroundColor: COLORS.white,
    alignSelf: 'center',
  },
  inActiveRadioButton: {
    width: 15,
    height: 15,
  },
});
