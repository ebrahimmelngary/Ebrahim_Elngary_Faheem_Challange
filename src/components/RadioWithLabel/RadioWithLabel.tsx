import React, { FC } from 'react';
import { Pressable, Text, View } from 'react-native';

import styles from './RadioWithLabel.styles';

type RadioWithLabelProps = {
  title: string;
  isActive: boolean;
  onPress: () => void;
};

const RadioWithLabel: FC<RadioWithLabelProps> = ({
  title,
  isActive,
  onPress,
}: RadioWithLabelProps) => {
  return (
    <Pressable style={styles.container} onPress={onPress}>
      <Text>{title}</Text>
      <View style={styles.radioButton}>
        <View
          style={
            isActive ? styles.activeRadioButton : styles.inActiveRadioButton
          }
        />
      </View>
    </Pressable>
  );
};

export default RadioWithLabel;
