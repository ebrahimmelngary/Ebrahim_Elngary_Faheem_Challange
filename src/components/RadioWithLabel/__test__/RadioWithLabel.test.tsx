import React from 'react';
import { render, cleanup } from '@testing-library/react-native';

import { Result } from '../../../types';
import RadioWithLabel from '../RadioWithLabel';

describe('AddressBookCard', () => {
  let props: JSX.IntrinsicAttributes & {
    title: string;
    isActive: boolean;
    onPress: () => void;
  };

  afterEach(cleanup);
  it('should match snapshot', () => {
    const { toJSON } = render(<RadioWithLabel {...props} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
