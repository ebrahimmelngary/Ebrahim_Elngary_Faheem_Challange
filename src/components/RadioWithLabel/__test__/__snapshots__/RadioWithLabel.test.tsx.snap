// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddressBookCard should match snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <Text />
  <View
    style={
      {
        "alignItems": "center",
        "borderColor": "#ffffff",
        "borderRadius": 25,
        "borderWidth": 3,
        "height": 25,
        "justifyContent": "center",
        "marginHorizontal": 10,
        "width": 25,
      }
    }
  >
    <View
      style={
        {
          "height": 15,
          "width": 15,
        }
      }
    />
  </View>
</View>
`;
